gooee/client.py

```diff
@@ -59,20 +60,32 @@ def _request(self, method, path, headers=None, data=None, params=None):
 
         return response
 
-    def authenticate(self, username, password):
-        """Authenticate to get the JWT token and stash it for use with future
-        requests.
-        """
-        payload = {
-            'username': username,
-            'password': password,
-        }
-        response = self.post('/auth/login', data=payload)
+    def authenticate(self, username=None, password=None, api_token=None):
+        """Assert and store API authentication credentials for future requests."""
+        # Authenticate with a username and password for a JWT token.
+        if username and password:
+            payload = {
+                'username': username,
+                'password': password,
+            }
+            response = self.post('/auth/login', data=payload)
+
+            # If the status was all good, stash the JWT token.
+            if response.status_code != 200:
+                raise GooeeException('Could not authenticate with the API username and password')
```

Comment:
not sure if this message is user facing (please ignore if it is) but in case it is maybe the word "API" can be dropped?

Reply:
This could potentially be User facing, yes.  I included it to make it more clear when we failed with an API Token versus the standard credentials.